graph-gateway/tests/it_network_subgraph_client.rs

```diff
@@ -0,0 +1,110 @@
+use std::time::Duration;
+
+use assert_matches::assert_matches;
+use graph_gateway::network::subgraph::Client;
+use thegraph_core::client::Client as SubgraphClient;
+use url::Url;
+
+/// Test helper to get the test url from the environment.
+fn test_base_url() -> Url {
+    std::env::var("IT_TEST_HOSTED_SERVICE_URL")
+        .expect("Missing IT_TEST_HOSTED_SERVICE_URL")
+        .parse()
+        .expect("Invalid IT_TEST_HOSTED_SERVICE_URL")
+}
+
+/// Test helper to build the subgraph url with the given subgraph ID.
+fn hosted_service_url_with_subgraph_name(name: impl AsRef<str>) -> Url {
+    test_base_url()
+        .join(&format!("subgraphs/name/{}", name.as_ref()))
+        .expect("Invalid URL")
+}
+
+/// The Graph Network Mainnet in the hosted service.
+///
+/// https://thegraph.com/hosted-service/subgraph/graphprotocol/graph-network-mainnet
+const GRAPH_NETWORK_MAINNET_SUBGRAPH_ID: &str = "graphprotocol/graph-network-mainnet";
```

Comment:
I agree, but unfortunately, that would be unproductive at this moment. IIRC, our queries failed the last time I tested against the arbitrum graph-network subgraph because the subgraph schema was not up-to-date.

In any case, I aim to have a replacement for the current implementation in this PR. The tests can be changed to run against the arbitral subgraph by changing the env variable name.

Another possibility is to develop a solution based on cargo features that allow switching the target without duplicating the files using some `#[cfg(...)]` attributes.

Reply:
That shouldn't be a problem anymore. The arbitrum subgraph is what we've been using in production for some time now.

Reply:
Cool! Now it works 💯 I have updated the tests to use the _Arbitrum Gateway_ URL and credentials.